src/app/app.component.spec.ts

```diff
@@ -1,27 +1,49 @@
-import { TestBed, async } from '@angular/core/testing';
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { TestBed, async, ComponentFixture } from '@angular/core/testing';
+import { MatIconRegistry } from '@angular/material';
+import { AuthService } from './auth.service';
+
 import { AppComponent } from './app.component';
+
+
 describe('AppComponent', () => {
+  let fixture: ComponentFixture<AppComponent>;
+  let comp: AppComponent;
+  let AuthServiceStub: Partial<AuthService>;
+
+  AuthServiceStub = {};
+
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       declarations: [
-        AppComponent
+        AppComponent,
       ],
+      schemas: [
+        CUSTOM_ELEMENTS_SCHEMA,
+      ],
+      providers: [
+        MatIconRegistry,
+        {provide: AuthService, useValue: AuthServiceStub },
+      ]
     }).compileComponents();
+
+    fixture = TestBed.createComponent(AppComponent);
+    comp = fixture.debugElement.componentInstance;
   }));
+
   it('should create the app', async(() => {
-    const fixture = TestBed.createComponent(AppComponent);
-    const app = fixture.debugElement.componentInstance;
-    expect(app).toBeTruthy();
+    expect(comp).toBeTruthy();
   }));
-  it(`should have as title 'app'`, async(() => {
-    const fixture = TestBed.createComponent(AppComponent);
-    const app = fixture.debugElement.componentInstance;
-    expect(app.title).toEqual('app');
+
+  it(`should have as title 'myBudget'`, async(() => {
+    expect(comp.title).toEqual('myBudget');
```

Comment:
Trovo un po' inutile un test sul titolo che deve essere uguale a 'myBudget'. Mi aspetto che il titolo possa cambiare da un monte all'altro, magari anche più volte nel corso dello sviluppo e quindi tutte le volte occorre cambiare anche il test per farlo passare.

Reply:
concordo che sia inutile testare una stringa, ma in questo caso vado a testare la proprieta' `title` della mia classe. vado a verificare che esista e che sia settata ad un valore in modo che non ritorni undefined o errori.